base/string.jl

```diff
@@ -968,8 +968,10 @@ byte_string_classify(s::ByteString) = byte_string_classify(s.data)
     # 1: valid ASCII
     # 2: valid UTF-8
 
-is_valid_ascii(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
-is_valid_utf8(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(::Type{ASCIIString}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
+isvalid(::Type{UTF8String}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(str::ASCIIString) = byte_string_classify(str) == 1
```

Comment:
Seems like it would be cleaner to have a single method:
```jl
isvalid{T<:Union(ASCIIString,UTF8String,UTF16String,UTF32String)}(s::T) = isvalid(T, s.data)
```